satpy/readers/seviri_base.py

```diff
@@ -193,11 +193,28 @@
 
 
 def get_cds_time(days, msecs):
-    """Get the datetime object of the time since epoch given in days and
-    milliseconds of day
+    """Compute timestamp given the days since epoch and milliseconds of the day
+
+    Args:
+        days (int or numpy.ndarray):
+            Days since 1958-01-01
+        msecs (int or numpy.ndarray):
+            Milliseconds of the day
+
+    Returns:
+        numpy.datetime64: Timestamp(s)
+
     """
-    return datetime(1958, 1, 1) + timedelta(days=float(days),
-                                            milliseconds=float(msecs))
+    if isinstance(days, (int, float)):
+        days = np.array([days], dtype='int64')
```

Comment:
What if days is a float ? Shouldn't something go over to msecs ?

Reply:
Right, that doesn't make sense. Both days and msecs are expected to be integer, so the check for float should be removed. It was meant to check whether the input is scalar. But maybe `np.isscalar` is a better option